### PR TITLE
Add keyrings.google-artifactregistry-auth

### LINKED ISSUE #32123: [Feature Request]: Include keyrings.google-artifactregistry-auth in Python SDK container
### What would you like to happen?

I want to use Python packages in private repository in  [Google Artifact Repository](https://cloud.google.com/artifact-registry/docs/python/authentication. It requires `keyrings.google-artifactregistry-auth` to run `pip install <package> --extra-index-url <location>-python.pkg.dev/<project>/<repository>/simple`. 

To run Dataflow job with private packages, currently I should create a custom container that installs `keyrings.google-artifactregistry-auth`. [1] is an example of Dockerfile for this. 

For convenience, can we pre-install the package in Python SDK containers?

[1]
```
FROM apache/beam_python3.11_sdk:2.58.0

RUN pip install keyrings.google-artifactregistry-auth
```

### Issue Priority

Priority: 2 (default / most feature requests should be filed as P2)

### Issue Components

- [X] Component: Python SDK
- [ ] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [ ] Component: IO connector
- [ ] Component: Beam YAML
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Infrastructure
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

### PR DESCRIPTION
**Pull request by baeminbo**
Fix #32123.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [x] Update `CHANGES.md` with noteworthy changes.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by baeminbo** [comment]
R: @damccorm

File: sdks/python/container/base_image_requirements_manual.txt [hunk 1]
```diff
@@ -38,4 +38,7 @@ nose==1.3.7  # For Dataflow internal testing. TODO: remove this.
 python-snappy  # Optimizes execution of some Beam codepaths.
 scipy
 scikit-learn
-build>=1.0,<2 # tool to build sdist from setup.py in stager.
\ No newline at end of file
+build>=1.0,<2 # tool to build sdist from setup.py in stager.
+# Authentication for Google Artifact Registry when using --extra-index-url or --index-url in requirements.txt in
+# Dataflow, which allows installing python packages from private Python repositories in GAR.
+keyrings.google-artifactregistry-auth
```
Comment (Abacn) [inline sdks/python/container/base_image_requirements_manual.txt:44]: I'm good with the change, just note another choice will be add this dependency to [gcp] extension in setup.py. Is there a preference here?

Comment (damccorm) [reply sdks/python/container/base_image_requirements_manual.txt:44]: +1 - I think adding it to the gcp extension ([here](https://github.com/apache/beam/blob/a7852d9ddb1726838c5878601c582fae45a8798b/sdks/python/setup.py#L428)) is probably a slightly better option.

**Review by damccorm — CHANGES_REQUESTED** [review]
I'm on board with the change overall, but would like us to incorporate Yi's suggestion

**Comment by damccorm** [comment]
Thanks for doing this!

File: sdks/python/container/base_image_requirements_manual.txt (hunk 1 above)
Comment (baeminbo) [reply sdks/python/container/base_image_requirements_manual.txt]: Thanks! I moved it to [gcp] extension in setup.py.

**Comment by Abacn** [comment]
yaml xlang direct precommit failed twice, seen this test flaky in general, merging for now

**Review by Abacn — APPROVED** [review]

**Pull request merged**